src/common/class-syntax.k

```diff
@@ -30,22 +31,15 @@ syntax ClassMemberDec ::= FieldDec | ClassDec | MethodDec | InterfaceDec | SemiC
 syntax SemiColon ::= ";" [klabel('SemiColon)]
 
 //@ \subsection{ConstructorDeclarations.sdf}
-//todo optionals
-syntax ConstrHead   ::=  AnnoConstrModList TypeParams Type Id "(" Params ")" Throws [prefer, klabel('ConstrDecHead)]
-                       | AnnoConstrModList TypeParams Type Id "(" Params ")"    [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")" Throws   [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")"    [klabel('ConstrDecHead)]
+syntax ConstrHead   ::=  AnnoConstrModList OptionalTypeParams Type
+                        Id "(" Params ")" OptionalThrows [klabel('ConstrDecHead)]
```

Comment:
Just for readability issues, you could write the entire production on the first line, and attributes on the second line.
At first I thought you forgot a vertical bar, but by looking at the deleted part I was able to understand the change. In the future I won't be able to read it that easy.